src/Runner.Worker/Container/DockerCommandManager.cs

```diff
@@ -346,6 +358,31 @@ public async Task<List<PortMapping>> DockerPort(IExecutionContext context, strin
             return DockerUtil.ParseDockerPort(portMappingLines);
         }
 
+        public async Task<int> DockerLogin(IExecutionContext context, string configFileDirectory, string registry, string username, string password)
+        {
+            string args = $"--config {configFileDirectory} login {registry} -u {username} --password-stdin";
+            context.Command($"{DockerPath} {args}");
+
+            var input = Channel.CreateBounded<string>(new BoundedChannelOptions(1) { SingleReader = true, SingleWriter = true });
+            input.Writer.TryWrite(password);
+
+            var processInvoker = HostContext.CreateService<IProcessInvoker>();
+
+            var task = processInvoker.ExecuteAsync(
```

Comment:
you can just `return processInvoker.ExecuteAsync()` and remove `async` from the method name.